app/code/core/Mage/Core/Block/Template.php

```diff
@@ -117,7 +117,7 @@ public function getTemplateFile()
         if ($area) {
             $params['_area'] = $area;
         }
-        return Mage::getDesign()->getTemplateFilename($this->getTemplate(), $params);
+        return Mage::getDesign()->getTemplateFilename((string)$this->getTemplate(), $params);
```

Comment:
```suggestion
        return (string)Mage::getDesign()->getTemplateFilename($this->getTemplate(), $params);
```
what if we do this instead? pros/cons?

Reply:
@fballiano nice thanks for your rapid feedback :-) Unfortunately this would not fix the issue because the `str_contains()` deprecation warning is in `Mage_Core_Model_Design_Package::getFilename()` which is called by `Mage_Core_Model_Design_Package::getTemplateFilename()`

Imho the optimum solution would be to change `getFilename()` and all related methods to use proper type hints but this would require rather more changes and therewith risk.

Reply:
mmm I see, but if $this->getTemplate() returns null, Mage::getDesign()->getTemplateFilename() would also not return anything right? if it is so, then maybe instead of calling Mage::getDesign()->getTemplateFilename() we should add an if/then statement?

Reply:
If `getTemplate()` returns `null`, `getTemplateFilename()` returns - apart from the warning - the path without the filename, e.g. `frontend/base/default//template/`.

With my proposed solution the logic/return values stay exactly the same, only the deprecation warning is fixed.